src/test/scala/com/campudus/tableaux/api/content/CellLevelAnnotationsTest.scala

```diff
@@ -337,4 +337,67 @@ class CellLevelAnnotationsTest extends TableauxTestBase {
       }
     }
   }
+
+  @Test
+  def addAnnotationWithoutLangtagsToConcatColumn(implicit c: TestContext): Unit = {
+    okTest {
+      for {
+        tableId <- createEmptyDefaultTable()
+
+        // make second column an identifer
+        _ <- sendRequest("POST", s"/tables/$tableId/columns/2", Json.obj("identifier" -> true))
+
+        // empty row
+        result <- sendRequest("POST", s"/tables/$tableId/rows")
+        rowId = result.getLong("id")
+
+        _ <- sendRequest("POST",
+          s"/tables/$tableId/columns/0/rows/$rowId/annotations",
+          Json.obj("type" -> "info", "value" -> "this is a comment"))
+
+        rowJson1 <- sendRequest("GET", s"/tables/$tableId/rows/$rowId")
+      } yield {
+        val exceptedColumn1Flags = Json.arr(Json.obj("type" -> "info", "value" -> "this is a comment"))
+
+        // we do have three columns because of the implicit concat column
+        assertEquals(3, rowJson1.getJsonArray("annotations").size())
+
+        // get annotation from concat cell
+        assertContainsDeep(exceptedColumn1Flags, rowJson1.getJsonArray("annotations").getJsonArray(0))
+
+        // no annotation on other cells
+        assertNull(rowJson1.getJsonArray("annotations").getJsonArray(1))
+        assertNull(rowJson1.getJsonArray("annotations").getJsonArray(2))
```

Comment:
Can we `assertEquals` directly against the complete structure?

```js
[
  [{type:"info", value: "this is a comment"}],
  null,
  null
]
```

or in Scala:

```scala
Json.arr(
  Json.arr(Json.obj("type" -> "info, "value" -> "this is a comment")),
  null,
  null
)
```
Might make the code more readable (but could be added to the "refactoring tests" task)

Reply:
Will do